File: p231e/power_of_two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function (n) {
  return n > 0 && n - (n & -n) === 0;
};
// TEST
[
  [0, false],
  [1, true],
  [2, true],
  [5, false],
  [10, false],
  [16, true],
  [218, false],
].forEach(([n, expected]) => {
  const actual = isPowerOfTwo(n);
  console.log('Is', n, 'power of 2? ->', actual);
  console.assert(actual == expected);
});

File: p231e/power_of_two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and n - (n & -n) == 0


# TESTS
for n, expected in [
    (0, False),
    (1, True),
    (2, True),
    (5, False),
    (10, False),
    (16, True),
    (218, False),
]:
    sol = Solution()
    actual = sol.isPowerOfTwo(n)
    print("Is", n, "power of two? ->", actual)
    assert actual == expected
